tests/meson.build

```diff
@@ -0,0 +1,74 @@
+# Test sources
+test_src = files(
+    'test_CommChannel.cc',
+    'test_Context.cc',
+    'test_exceptions.cc',
+    'test_Executor.cc',
+    #'tests/test_format.cc' needs fmt{} library
+    'test_LockedQueue.cc',
+    'test_main.cc',
+    'test_Message.cc',
+    'test_Sequence.cc',
+    'test_SequenceManager.cc',
+    'test_serialize_sequence.cc',
+    'test_Step.cc',
+    'test_time_types.cc',
+    'test_VariableName.cc',
+)
+
+# The tests are executed in the build dir to avoid pollution of the git repository with
+# the test executable's output files
+test('all',
+    executable(meson.project_name() + '_test',
+        test_src,
+        dependencies : taskolib_dep,
+    ),
+    workdir : meson.current_build_dir(),
+    timeout : 10,
+)
+
+fmt_dep = dependency('fmt', required : false)
+if fmt_dep.found()
+    test('format',
+        executable(meson.project_name() + '_format_test',
+            [ 'test_format.cc', 'test_main.cc', ],
+            dependencies : [ taskolib_dep, dependency('fmt'), ],
+        ),
+        timeout : 10,
+    )
+endif
+
+######
+# Test if all headers are self-contained (Core Guidelines SF.11)
+# This is automated over all public_headers
+
+message('Generating self-containment tests')
+all_header_checks = [ ]
+loop_number = 0
+foreach header_to_check : public_headers
+    loop_number += 1
+    if header_to_check.startswith('include/') and header_to_check.endswith('.h')
+        header_to_check = header_to_check.substring(8)
+        header_name = header_to_check.split('.h')[0]
+        if not fmt_dep.found() and header_name.endswith('/format')
+            continue
+        endif
+
+        header_conf = configuration_data()
+        header_conf.set('HEADER_TO_INCLUDE', header_to_check)
+        header_conf.set('DUMMY_FUNCTION_NAME', 'a@0@'.format(loop_number))
+
+        header_check = configure_file(
```

Comment:
As far as I can see it from meson [docu page](https://mesonbuild.com/Reference-manual_functions.html#configuration_data) it replaces from the previous created `HEADER_TO_INCLUDE` and `DUMMY_FUNCTION_NAME` configuration and creates `incl_<header_name_without_h>.cc` with function `void a<number>() {}`, correct?

Reply:
The `header_conf` configuration set does later on do string replacements.
`@HEADER_TO_INCLUDE@` is replaced by the string contained in `header_to_check`.
`@DUMMY_FUNCTION_NAME@` is replaced by `a1` (or whatever the loop_number is at that point).

The replacement takes place on `test_self_containment.cc.in` 
```
include "@HEADER_TO_INCLUDE@"

void @DUMMY_FUNCTION_NAME@() {} // to prevent object files without any symbol
```
and creates a temporary `incl_*cc` as you said.

You can see these files after building in the `build/tests/` directory.

For each public header we create a `*.cc` file that includes just that one header and nothing else (and has an empty function so that an object will be created).